OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -54,22 +55,24 @@ public class PathFinder : IPathFinder
 		static readonly List<CPos> EmptyPath = new List<CPos>(0);
 		readonly World world;
 		DomainIndex domainIndex;
-		bool cached;
+		bool domainIndexCached;
+		readonly PathCacheStorage cacheStorage;
 
-		public PathFinder(World world)
+		public PathFinder(World world, PathCacheStorage cacheStorage)
 		{
 			this.world = world;
+			this.cacheStorage = cacheStorage;
```

Comment:
Style nit: can we initialize this as `cacheStorage = new PathCacheStorage(World)` instead of passing it in?